### PR TITLE
crypto/homo/cl: check that the generator g is non-identity

### PR DESCRIPTION
**Pull request by cychuang0924**
**Before submitting a pull request**, please make sure the following is done:

### Checklist

- [x] Fork [the repository](https://github.com/getamis/alice) and create your new branch from `master`.
- [x] Please mention the name of community in Pull Request title.
- [x] Git Commit Messages
    - [x] Use the present tense (Also in Pull Request title): "Add feature" not "Added feature"
    - [x] Use the imperative mood (Also in Pull Request title): "Move cursor to..." not "Moves cursor to..."
    - [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.

### CODE REVIEW

File: crypto/homo/cl/cl.go [hunk 1]
```diff
@@ -534,3 +537,22 @@ func generateAnotherPrimeQ(p *big.Int, bitsQ int) (*big.Int, error) {
 		return q, nil
 	}
 }
+
+func getNonIdentityGenerator(generator *bqForm.BQuadraticForm, upperBound *big.Int) (g *bqForm.BQuadraticForm, err error) {
```
Comment (markya0616) [inline crypto/homo/cl/cl.go]: Don't use named return

File: crypto/homo/cl/cl.go [hunk 2]
```diff
@@ -534,3 +537,22 @@ func generateAnotherPrimeQ(p *big.Int, bitsQ int) (*big.Int, error) {
...
+		b, err = utils.RandomPositiveInt(upperBound)
+		if err != nil {
+			return nil, err
+		}
+		g, err = generator.Exp(b)
+		if err != nil {
+			return nil, err
+		}
+		if !g.Equal(identity) {
+			return
```
Comment (markya0616) [inline crypto/homo/cl/cl.go]: return g, nil

File: crypto/homo/cl/cl.go [hunk 3]
```diff
@@ -534,3 +537,22 @@ func generateAnotherPrimeQ(p *big.Int, bitsQ int) (*big.Int, error) {
 		return q, nil
 	}
 }
+
+func getNonIdentityGenerator(generator *bqForm.BQuadraticForm, upperBound *big.Int) (g *bqForm.BQuadraticForm, err error) {
+	var b *big.Int
```
Comment (markya0616) [inline crypto/homo/cl/cl.go]: rm

File: crypto/homo/cl/cl.go [hunk 4]
```diff
@@ -534,3 +537,22 @@ func generateAnotherPrimeQ(p *big.Int, bitsQ int) (*big.Int, error) {
...
+	identity := generator.Identity()
+	for i := 0; i < maxGenG; i++ {
+		b, err = utils.RandomPositiveInt(upperBound)
```
Comment (markya0616) [inline crypto/homo/cl/cl.go]: b, err :=

File: crypto/homo/cl/cl.go (hunk 1 above)
Comment (cychuang0924) [reply crypto/homo/cl/cl.go]: done

File: crypto/homo/cl/cl.go (hunk 2 above)
Comment (cychuang0924) [reply crypto/homo/cl/cl.go]: done

File: crypto/homo/cl/cl.go (hunk 3 above)
Comment (cychuang0924) [reply crypto/homo/cl/cl.go]: done

File: crypto/homo/cl/cl.go (hunk 4 above)
Comment (cychuang0924) [reply crypto/homo/cl/cl.go]: done

**Pull request merged**